vic/vic_run/src/func_surf_energy_bal.c

```diff
@@ -749,67 +743,46 @@ func_surf_energy_bal(double  Ts,
 
        Should evapotranspiration be active when the
        ground is only partially covered with snow????
-
-       Use Arno Evap in the exposed soil portion, and/or
-       if LAI is zero.
     *************************************************/
-    if (VEG && !SNOWING && veg_var->fcanopy > 0) {
-        Evap = canopy_evap(layer, veg_var, true,
-                           veg_class, Wdew, delta_t, NetBareRad, vpd,
-                           NetShortBare, Tair, Ra_veg[1], elevation, rainfall,
-                           Wmax, Wcr, Wpwp, frost_fract, root, dryFrac,
-                           shortwave, Catm, CanopLayerBnd);
-        if (veg_var->fcanopy < 1) {
-            for (i = 0; i < options.Nlayer; i++) {
-                transp[i] = layer[i].evap;
-                layer[i].evap = 0.;
-            }
+    for (i=0; i<options.Nlayer; i++) {
+        layer[i].transp = 0;
+        layer[i].esoil = 0;
+    }
+    Evap = 0.;
+    if (!SNOWING) {
+        if (VEG && veg_var->fcanopy > 0) {
+            Evap = canopy_evap(layer, veg_var, true, veg_class, Wdew,
```

Comment:
I'm not taking issue with this, but I don't understand the motivation behind separating out the `(!SNOWING)` from the `VEG` and `veg_var->fcanopy` into separate lines?

Reply:
I minimized redundant code.  If you start with a condition that links the 3 conditions `VEG`, `!SNOWING`, and `fcanopy > 0`, you'll necessarily need further compound conditions later on to handle all possible cases.  The most important case is simply `!SNOWING`.  If it is snowing, `Evap` is `0`.  So, I separated that from the other conditions.  Then, the nested check on `VEG` and `fcanopy > 0` determines whether we call `canopy_evap()` and how `SurfRad` is set. Note: I'm now removing lines 782-784, since they were redundant with my initialization of `Evap` to 0 on line 751.

Reply:
Got it - thanks for clarifying